boggle_blast.lic

```diff
@@ -0,0 +1,165 @@
+
+class BoggleBlast
+  include DRC
+  include DRCH
+  include DRCI
+  include DRCM
+  include DRCT
+
+  def initialize
+    @stow_container = "backpack" # set this to be your main default storage container - i.e. the one used in STORE LIST set to default
+    @full_container = "haversack" # this container is used to store full 100k HE ticket stacks, set it to a DIFFERENT container than the default
+    @currency = "dokoras" # don't change this
+    @withdraw = true # change to false if you don't want to withdraw more money when you run out
+    @portal_room = 2780 # 2780 = Shard, 932 = Crossing, no clue about the others
+    if DRSkill.getrank('Slings') >= 500
+      @cash = 500 # change to whatever amount you want to withdraw at a time
+      @denomination = "platinum" # change to whatever valid denomination you'd like
+    else
+      @cash = 5 # This is the amount used if you have an elevated risk of injury or death due to low skill, suggest keeping it low
+      @denomination = "platinum" # change to whatever valid denomination you'd like
+    end
+    @trash_items = ['robe', 'blouse', 'pajamas', 'cambrinth armband', 'pants', 'anklet', 'towel', 'keyblank', 'blossom', 'amulet', 'greatcloak', 'cloak', 'coat', 'shirt', 'pumpkins', 'mirror', 'globe', 'earrings', 'wings', 'compass', 'cambrinth pumpkin', 'spectacles', 'jar', 'clay', 'fabric', 'strands', 'flamewood case', 'boggle ring', 'leather boots', 'tomiek-rimmed goggles', 'scarecrow doll', 'boggle doll', 'heart', 'garnet', 'comb', 'tomiek ring', 'animite ring', 'wooden earcuff', 'orange hat', 'wooden armband', 'gem pouch', 'sling', 'seeds', 'purse', 'simple necklace', 'electrum charm', 'glass pestle', 'cookie', 'candy', 'pumpkin pie', 'mortar', 'ring', 'statuette', 'lollipop', 'trousers', 'rocks', 'carving knife', 'beer', 'pendant', 'torque', 'rose'] # this is my list of trash items, add/remove what nouns you like
+    @danger_items = ['doll'] # todo: add this in to trash_items?
+    @fake_pumpkins = ['wooden pumpkins', 'cambrinth pumpkin']
+    do_boggle_crap
+  end
+
+  def do_boggle_crap
+    loop do
+      heal_up
+      stow_hands
+      ditch_pumpkins
+      empty_hands
+      check_money
+      go_play
+      check_prize
+    end
+  end
+
+  def heal_up
+    check_health
+    if bleeding?
+      walk_to('19104')
+      bput("go path", "Obvious paths: none")
+      pause 5 # assumes quick spider travel option purchased
+      wait_for_script_to_complete('safe-room')
+      pause
+      walk_to(@portal_room)
+      bput("go path", "Obvious paths: none")
+      pause 5 # assumes quick spider travel option purchased
+      walk_to('10584')
+      pause
+    end
+  end
+
+  def check_money
+    get_money if check_wealth('dokoras') < 200 && @withdraw
+  end
+
+  def get_money
+    walk_to('10358')
+    case bput("withdraw #@cash #@denomination #@currency ", 'The clerk counts out', 'The clerk tells you', 'You must be at')
```

Comment:
❌ This looks wrong. Should it be `bput("withdraw #{@cash} #{@denomination} #{@currency}"`? (The string interpolation is missing)

Reply:
Ruby allows you to do #@[variable] interpretation, it's terrible and we shouldn't use it, but it allows it.

>When the expression to be interpolated into the string literal is simply a reference to a global, instance or class variable, then the curly braces may be omitted.

Reply:
Fascinating. I hate it, but if it works, I'll take the script.

@bmcneff consider these comments 💡 s then.